app/src/lib/menu-item.ts

```diff
@@ -72,3 +74,58 @@ export function getPlatformSpecificNameOrSymbolForModifier(
   // Not a known modifier, likely a normal key
   return modifier
 }
+
+/** Show the given menu items in a contextual menu. */
+export async function showContextualMenu(
+  items: ReadonlyArray<IMenuItem>,
+  addSpellCheckMenu = false
+) {
+  const indices = await invokeContextualMenu(
+    serializeMenuItems(items),
+    addSpellCheckMenu
+  )
+
+  if (indices !== null) {
+    const menuItem = findSubmenuItem(items, indices)
+
+    if (menuItem !== undefined && menuItem.action !== undefined) {
+      menuItem.action()
+    }
+  }
+}
+
+/**
+ * Remove the menu items properties that can't be serializable in
+ * order to pass them via IPC.
+ */
+function serializeMenuItems(
```

Comment:
`serializeMenuItems` and `findSubmenuItem` are pretty much lift and shift (just a documentation change on on).